proof_of_concept/ddm_site.py

```diff
@@ -48,7 +51,15 @@ def __init__(
         self.administrator = owner
         self.namespace = namespace
 
-        self._ddm_client = DDMClient(self.name)
+        # Load API definitions
+        site_api_file = Path(__file__).parent / 'site_api.yaml'
+        with open(site_api_file, 'r') as f:
+            site_api_def = yaml.safe_load(f.read())
+
+        self._site_validator = Validator(site_api_def)
```

Comment:
There is a lot of passing around of this site_validator objects which just depends on a static file, it almost feels like a constant. In the end we only use it to init a PolicyClient and a DDMClient. This was kind of confusing for me, because it was hard to track why we would need to pass around this thing that is only needed to validate API input/output. I think the declaration of it in DDMClient made sense, maybe we can declare it in PolicyClient as well? Or just have a module `validation.py` and move all validator declerations there?

Reply:
Hmm. I'm not generally a fan of singletons (and they're considered a bad idea in general as well), but the fact that it's basically just a wrapper around the YAML file may tip the balance here. I don't like duplicating the code, but having a static object in validation.py may make sense (and then later a rest/validation/ directory with the YAML files and validation.py), and you're right that lugging this thing around everywhere is also not optimal. Let me sleep on it for a bit :-).

Reply:
I suggest maybe making a separate issue for it, I see that also in in the other PRs you use this pattern. And then you can  just take the time to sleep on it :)

Reply:
Then you can go ahead and merge this.

Reply:
Some useful background: https://docs.microsoft.com/en-gb/archive/blogs/scottdensmore/why-singletons-are-evil